railties/test/application/test_runner_test.rb

```diff
@@ -651,6 +651,45 @@ class PostTest < ActiveSupport::TestCase
       end
     end
 
+    def test_declarative_style_with_tests_that_use_minitest_spec
+      app_file "test/models/post_test.rb", <<~RUBY
+        require "test_helper"
+
+        class PostTest < ActiveSupport::TestCase
+          define_method "test_foo" do
+            puts "hello foo"
+            assert true
+          end
+
+          define_method "test_foo  again" do
+            puts "hello again"
+            assert true
+          end
+
+          define_method "test_foo no more" do
+            flunk
+          end
+        end
+      RUBY
+
+      run_test_command("test/models/post_test.rb -n 'foo'").tap do |output|
+        assert_match "hello foo", output
+        assert_match "1 runs, 1 assertions, 0 failures", output
+      end
+
+      run_test_command("test/models/post_test.rb -n 'foo  again'").tap do |output|
+        assert_match "hello again", output
+        assert_match "1 runs, 1 assertions, 0 failures", output
+      end
```

Comment:
These assertions pass, but I don't think `-n 'foo  again'` would actually work with `Minitest::Spec::it` because, as you pointed out, it [adds a number](https://github.com/minitest/minitest/blob/a9fa045044b4210cfd21a512b06d1a4527d709ba/lib/minitest/spec.rb#L226-L229) after the `test_` prefix.  So the test name for `foo  again` would actually be `test_0003_foo  again`, which the `/^test_#{new_filter}$/` regexp does not match.

However, if my understanding is correct, `-n 'foo  again'` did not work before #44290 either, did it?  Only `-n '/foo  again/'` worked, is that correct?

I think the only reliable way to make `-n 'foo  again'` work would be to add something like `(?:_\d+)?` to the regexp, but I'm not sure if we want to go that far.  On the other hand, making `-n '/foo  again/'` work again requires a less-specific change, so I think that is reasonable.

I am pushing a commit to show what I mean.  The commit also changes this test to explicitly use `Minitest::Spec::it`, to be sure that it works.  Let me know what you think.

Reply:
> However, if my understanding is correct, `-n 'foo again'` did not work before #44290 either, did it? Only `-n '/foo again/'` worked, is that correct?

I think so. I came across this issue when trying to run tests using the [`m` gem](https://github.com/qrush/m), which [always provides a regex](https://github.com/qrush/m/blob/c86bf774f7daa5c3ea32e1f9597171fe30004e3e/lib/m/executor.rb#L23). So I don't have first hand knowledge of what does/doesn't work when trying to run tests by an exact name. But what you are saying seems correct.

Reply:
> The commit also changes this test to explicitly use `Minitest::Spec::it`, to be sure that it works.

That's a great idea! Thanks for doing that. 

The commit looks good. Just two notes:

1) The ever-expanding regex issue mentioned above - just worth checking that doesn't happen here.
2) Does it work as expected if there are multiple spaces in the test name?

Reply:
> The ever-expanding regex issue mentioned above - just worth checking that doesn't happen here.

Taken care of by using `gsub(/\s+/, ...)`, which is idempotent as long as `...` does not include whitespace. :+1:

> Does it work as expected if there are multiple spaces in the test name?

Yes.  I have added test coverage for it in the most recent commit. :+1: